WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt

```diff
@@ -510,39 +512,46 @@ class StoryComposerActivity : ComposeLoopFrameActivity(),
+            val assignedTempId = saveStoryGutenbergBlockUseCase.getTempIdForStoryFrame(
+                    newTempId, storyIndex, frameIndex
+            )
             when (frame.id) {
                 // if the frame.id is null, this is a new frame that has been added to an edited Story
                 // so, we don't have much information yet. We do have the background source (not the flattened
                 // image yet) so, let's use that for now, and assign the temporaryID we'll use to send
                 // save progress events to Gutenberg.
                 null -> {
-                    val storyMediaFileData =
-                            saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryIdNoMediaFile(
-                                    temporaryId = "$storyIndex-$frameIndex",
-                                    url = if (frame.source is FileBackgroundSource) {
-                                        (frame.source as FileBackgroundSource).file.toString()
-                                    } else {
-                                        (frame.source as UriBackgroundSource).contentUri.toString()
-                                    },
-                                    isVideo = (frame.frameItemType is VIDEO)
-                            )
+                    val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                            frame,
+                            assignedTempId
+                    )
                     frame.id = storyMediaFileData.id
                     storyMediaFileDataList.add(storyMediaFileData)
                 }
-                // if the frame.id is populated, this should be an actual MediaModel mediaId so,
+                // if the frame.id is populated and is not a temporary id, this should be an actual MediaModel mediaId so,
                 // let's use that to obtain the mediaFile and then replace it with the temporary frame.id
                 else -> {
                     frame.id?.let {
-                        val mediaModel = mediaStore.getSiteMediaWithId(site, it.toLong())
-                        val mediaFile = fluxCUtilsWrapper.mediaFileFromMediaModel(mediaModel)
-                        mediaFile?.let {
-                            val storyMediaFileData =
-                                    saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
-                                            mediaFile = it,
-                                            temporaryId = "$storyIndex-$frameIndex"
-                                    )
-                            frame.id = storyMediaFileData.id
+                        if (it.startsWith(TEMPORARY_ID_PREFIX)) {
+                            val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                                    frame,
+                                    it
+                            )
+                            // frame.id = storyMediaFileData.id
```

Comment:
hi @mzorz do we want to remove this line or is it going to be utilized in the future?

Reply:
nice catch! removed in 5e55e168f6